### ISSUE TITLE
Typescript issue - Property 'truncated' does not exist on type 'Readable'

### ISSUE THREAD

**Issue by sanzhardanybayev** [fastify/fastify-multipart#272]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

Whenever I try to access **truncated** property I get the following TS error - **Property 'truncated' does not exist on type 'Readable'.**

Here is the example: 
```js
const data = await req.file()
await pump(data.file, fs.createWriteStream(data.filename))
if (data.file.truncated) {
  // you may need to delete the part of the file that has been saved on disk
  // before the `limits.fileSize` has been reached
  reply.send(new fastify.multipartErrors.FilesLimitError());    
}
```

**Comment by mcollina** [comment]
Can you please include a full example to reproduce?

**Comment by sanzhardanybayev** [comment]
@mcollina yeah, sure.
I'm using NestJS under the hood. 

Here's controller's code :
```js
import { Controller, Get, Post, Req, Res } from '@nestjs/common';
import { pipeline, Readable } from 'stream';
import * as util from 'util';
import { FastifyReply, FastifyRequest } from 'fastify';
import { InjectModel } from '@nestjs/mongoose';
import { Model } from 'mongoose';

import { Request } from './interfaces/request';
import * as fs from 'fs';
import { MultipartFile, MultipartValue } from 'fastify-multipart';

const pump = util.promisify(pipeline);

@Controller('request')
export class RequestController {
  constructor(
    @InjectModel('Request') private readonly request: Model<Request>,
  ) {}

  @Get('/')
  async getRequests(@Res() reply: FastifyReply) {
    const request = await this.request.find().limit(1);
    reply
      .code(200)
      .header('Content-Type', 'application/json; charset=utf-8')
      .send({ request });
  }

  @Post('/create')
  async createRequest(@Req() req: FastifyRequest, @Res() reply: FastifyReply) {

    const data = await req.file()
    await pump(data.file, fs.createWriteStream(data.filename))
    if (data.file.truncated) {
      // you may need to delete the part of the file that has been saved on disk
      // before the `limits.fileSize` has been reached
      reply.send('Error occurred');
    }

    reply.status(200).send('Uploaded');
  }
}
```

**Comment by mcollina** [comment]
Could you create an example just using a self-runnable Fastify instance? Thanks

**Comment by sanzhardanybayev** [comment]
@mcollina Here's the repo - https://github.com/sanzhardanybayev/nestjs-fastify-file-upload 

Just run `yarn && yarn start:dev` after cloning. 

The issue is the the following path - `./src/request/request.controller.ts`

**Comment by mcollina** [comment]
I might not have been clear: could you create a runnable example that does **not** have any other dependencies minus fastify and fastify-multipart?

**Comment by sanzhardanybayev** [comment]
@mcollina My bad! Well , I created and empty project with just **fastify** and **fastify-multipart** installed. Funny thing that this TS issue isn't shown anymore.  I guess this might be **nestjs** related issue.

**Comment by sanzhardanybayev** [comment]
For those who might get the same issue, here's the workaround :

```js
import { Controller, Get, Post, Req, Res } from '@nestjs/common';
import { pipeline, Readable } from 'stream';
import * as util from 'util';
import { FastifyReply, FastifyRequest } from 'fastify';
import { InjectModel } from '@nestjs/mongoose';
import { Model } from 'mongoose';

import { Request } from './interfaces/request';
import * as fs from 'fs';
import { MultipartFile } from 'fastify-multipart';

const pump = util.promisify(pipeline);

type Files = MultipartFile & {
  file: Readable & {
    truncated: boolean;
  };
};

@Controller('request')
export class RequestController {
  constructor(
    @InjectModel('Request') private readonly request: Model<Request>,
  ) {}

  @Get('/')
  async getRequests(@Res() reply: FastifyReply) {
    const request = await this.request.find().limit(1);
    reply
      .code(200)
      .header('Content-Type', 'application/json; charset=utf-8')
      .send({ request });
  }

  @Post('/create')
  async createRequest(@Req() req: FastifyRequest, @Res() reply: FastifyReply) {
    const parts = (await req.files({
      limits: { fileSize: 1 },
    })) as AsyncIterableIterator<Files>;

    for await (const part of parts) {
      await pump(part.file, fs.createWriteStream(part.filename));

      if (part.file.truncated) {
        console.log('TO BE REMOVED', part.filename);
        fs.unlink(part.filename, () => {
          console.log('Removed');
        });
      }
    }

    reply.send('Hello');
  }
}
```

What I basically did is created a custom type which extended **Readable** class  and used **type assertion**.

I tried do it it with **augmenting module via .d.ts files**, but it didn't work.

**Comment by Zhongy1** [comment]
I ran into the same problem. Diving into the type definitions, I found that the Readable class doesn't have a "truncated" property defined, which is resulting in this type issue. For a cleaner work around, I used this:
``` Typescript
declare module 'stream' {
    interface Readable {
        truncated: boolean
    }
}
```
This will allow you to use the truncated property without needing to do type assertions.
Be aware that this exposes the truncated property on all Readable objects imported from "stream."
